modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/definitions/display/context/util/ObjectCodeEditorUtil.java

```diff
@@ -222,7 +228,8 @@ public static List<HashMap<String, String>> getItems(Locale locale) {
 			List<HashMap<String, String>> values = new ArrayList<>();
 
 			for (DDMExpressionFunction ddmExpressionFunction : values()) {
-				if (StringUtil.equals(
+				if ((PropsValues.OBJECT_ENTRY_SCRIPT_VARIABLES_VERSION != 2) &&
+					StringUtil.equals(
 						ddmExpressionFunction._key, "old-value") &&
 					!GetterUtil.getBoolean(
 						PropsUtil.get("feature.flag.LPS-171440"))) {
```

Comment:
@pedro-oliveira446 @p-albuquerque if the feature flag is not enabled then we should keep hiding the new feature, even if the version is 2. Could you please double check this logic?

I think it should be something like:

```
StringUtil.equals(ddmExpressionFunction._key, "old-value") &&
(
	!GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-171440")) || 
	(PropsValues.OBJECT_ENTRY_SCRIPT_VARIABLES_VERSION != 2)
)
```